src/server/RHData.py

```diff
@@ -836,6 +843,23 @@ def reset_pilots(self):
         logger.info('Database pilots reset')
         return True
 
+    #Pilot Attributes
+    def get_pilot_attribute(self, pilot_or_id, name):
+        if isinstance(pilot_or_id, Database.Pilot):
+            pilot_id = pilot_or_id
+        else:
+            pilot_id = self._Database.Pilot.query.get(pilot_or_id).id
+
+        return self._Database.PilotAttribute.query.filter_by(id=pilot_id, name=name).one_or_none()
+
+    def get_pilot_attributes(self, pilot_or_id):
+        if isinstance(pilot_or_id, Database.Pilot):
+            pilot_id = pilot_or_id
+        else:
+            pilot_id = self._Database.Pilot.query.get(pilot_or_id).id
+
+        return self._Database.PilotAttribute.query.filter_by(id=pilot_id).all()
+
```

Comment:
These two methods look as if the can be implemented with one generic query builder method that does the whole isinstance and pilot_id stuff, and returns self._Database.PilotAttribute.query, so that all you do is implement the two differences. .filter_by(id=pilot_id).all(), and .filter_by(id=pilot_id, name=name).one_or_none()